text/0470-aurora-serverless-v2.md

```diff
@@ -0,0 +1,208 @@
+# Aurora Serverless v2 support
+
+* **Original Author(s):**: @pahud
+* **Tracking Issue**: #470
+* **API Bar Raiser**: @vinayak-kukreja
+
+Allow users to create Amazon Aurora Serverless v2 instances with the `aws-rds` module.
+
+## Working Backwards
+
+> This section should contain one or more "artifacts from the future", as if the
+> feature was already released and we are publishing its CHANGELOG, README,
+> CONTRIBUTING.md and optionally a PRESS RELEASE. This is the most important
+> section of your RFC. It's a powerful thought exercise which will challenge you
+> to truly think about this feature from a user's point of view.
+>
+> Choose *one or more* of the options below:
+>
+> * **CHANGELOG**: Write the changelog entry for this feature in conventional
+>   form (e.g. `feat(eks): cluster tags`). If this change includes a breaking
+>   change, include a `BREAKING CHANGE` clause with information on how to
+>   migrate. If migration is complicated, refer to a fictional GitHub issue and
+>   add its contents here.
+
+feat(rds): Aurora Serverless v2 support
+
+## Use Cases
+
+1. To create a new Aurora Serverless v2 cluster with one writer only. This writer could be either serverless or provisioned instance:
+
+- serverless writer
+- provisioned writer
+
+2. To create a new Aurora Serverless v2 cluster with one writer and one reader as serverless or provisoined. This could be:
+
+- serverless writer + serverless reader
+- serverless writer + provisioned reader
+- provisioned writer + serverless reader
+- provisioned writer + provisioned reader
+
+3. To create a new Aurora Serverless v2 cluster with one writer and multiple readers. This could be:
+
+- serverless writer + all serverless readders
+- serverless writer + all provisioned readers
+- serverless writer + mixed readers
+- provisioned writer + all serverless readers
+- provisioned writer + all provisioned readers
+- provisioned writer + mixed readers
```

Comment:
After syncing with the Aurora team, listing here for visibility what this design needs to look out for,
* Supporting a customer use case where the cluster could be empty i.e. without any instance and just the storage layer. This helps customers not actively using the database and remove the compute layer which adds costs for them. This could potentially also help in migrating from an existing cluster to a serverless v2 supported cluster. 
* Decide on a default scenario for provisioning.
* Decide on default minimum and maximum ACUs for serverless v2 instance. Should there be different defaults based on what database engine is selected?
* Helper methods to add/remove/upgrade instances.
* Metrics to detect failover in instances to alarm over and general database cluster metrics that can be helpful for the customers.

During implementation,
* Testing that provisioning is working with all supported database engines.
* Callouts in documentation about
  * Unreachable writer instance due to DNS caching if the instance was replaced due to a failover.
  * Cluster instances sizes are recommended to be similar to better handle a failover scenario.